doc/proposals/scorecard-custom-tests-2.md

```diff
@@ -0,0 +1,224 @@
+```
+apiVersion: apps/v1
+kind: Pod
+metadata:
+  name: isvtest1
+  labels:
+    vendor: someisv
+    test: isvtest1
+    suite: isvcustom
+```
+
+The end user could run that test with this command:
+```
+operator-sdk scorecard -o text --selector=test=isvtest1
+```
+
+#### Migrating Scorecard Internal Tests to Custom Test Images
+
+With the above custom test capability, there is benefit to Red Hat in migrating the existing scorecard internal tests to be based on the custom test format. 
+
+Today there are 2 internal scorecard tests that make use of the scorecard proxy, with these tests migrating to the custom test format, the scorecard proxy is likely not necessary which would also reduce the overall scorecard implementation complexity.
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+### Upgrade / Downgrade Strategy
+
+To implement this custom test functionality, additions or changes to the existing scorecard configuration file and command line flags would be necessary and would not be supported in previous scorecard releases.
+
+## Implementation History
+
+ * Proposal initiated in March 2020.
+
+## Drawbacks
+
+
+## Ansible and Helm Operator Support
+
+### Ansible Operator Complex Tests
+
+For custom tests, it might be possible to write the custom test image using ansible's callback plugin framework to produce container image log output in the scorecard required format (e.g. v1alpha2).  The complexity of the test might dictate whether this is viable or not, the fallback would be for the custom test to be developed in python, Java, or golang as discussed in this proposal.
+
+### Helm Operator Complex Tests
+
+Writing a custom test image in a Helm chart was not evaluated or pursued in this proposal.  What is likely for a Helm operator is that the operator owner would need to write custom test images using either Python, Java, golang, or some other programming language.
+
+
+## Alternatives
+
+The alternatives to this design proposal might include:
+ * Other existing/open-source test frameworks could be evaluated for use within the SDK for implementing the desired custom test functionality.
+
+
+## Conclusion
+
+The proposed changes to the scorecard solve the immediate need to support custom or user-provided tests, both simple and complex tests.  However, the longer term implication for scorecard is that all of its tests would evolve to use a single testing format that is far more flexible that what exists today in the current implementation.
+
+The proposed design focuses heavily on separation of concerns, turning scorecard into a test runner essentially, moving test implementations into their own concern (eg. container images).
+
+## Reference Material
+
+<https://github.com/kudobuilder/kuttl>
```

Comment:
Could we link to the original proposal as well?

Reply:
link provided in latest commit.